source/aws_solutions/qnabot/cli/qnabot_cli_helper.py

```diff
@@ -0,0 +1,446 @@
+        )
+        bucket_name = response["StackResourceDetail"]["PhysicalResourceId"]
+    except ClientError as err_exception:
+        return error_response(
+            error_code=err_exception.response["Error"]["Code"],
+            message=err_exception.response["Error"]["Message"],
+            comments="Please check the CloudFormation Stack being used is for a QnABot deployment, "
+            + "and that the stack has deployed successfully.",
+            status="Error",
+            show_error=True,
+        )
+    return bucket_name
+
+
+def initiate_import(
+    cloudformation_stack_name: str, source_filename: str, file_format: str, delete_existing_content: bool
+):
+    """
+    Initiate import process
+    :param bucket: Bucket to import to
+    :param source_filename: import directory and filename
+    :return: response status of the import request
+    """
+
+    importdatetime = datetime.datetime.utcnow()  # get current request date time in UTC timezone
+    # get Import bucket name from the cloudformation stack
+    str_import_bucket_name = get_bucket_name(cloudformation_stack_name, BucketType.IMPORT_BUCKET)
+
+    # create a options json config that includes import options that were used
+    str_import_options = {
+        "source_filename": source_filename,
+        "options": {"delete_existing_content": delete_existing_content, "file_format": file_format},
+        "import_datetime": str(importdatetime),
+        "source_application": "qnabot-cli",
+    }
+    str_import_options = json.dumps(str_import_options, indent=4)
+
+    try:  # put object in S3 bucket
+        s3_client = get_service_client("s3")  # boto3.client('s3')
+        # create a options json config file that includes import options that were used
+        response = s3_client.put_object(
+            Bucket=str_import_bucket_name, Key="options/" + os.path.basename(source_filename), Body=str_import_options
+        )
+
+        if file_format == "JSON":
+            str_file_contents = convert_json_to_jsonl(
+                source_filename
+            )  # convert to JSON Lines format (if input is JSON format)
+            # upload the contents of the converted json file to S3
+            response = s3_client.put_object(
+                Bucket=str_import_bucket_name, Key="data/" + os.path.basename(source_filename), Body=str_file_contents
+            )
+        else:
+            with open(source_filename, "rb") as obj_file:  # open file object
+                # upload the contents of the json file to S3
+                response = s3_client.put_object(
+                    Bucket=str_import_bucket_name, Key="data/" + os.path.basename(source_filename), Body=obj_file
+                )
+
+        # check status of the file import
+        response = get_import_status(
+            bucket=str_import_bucket_name, source_filename=source_filename, importdatetime=importdatetime
+        )
+
+        while json.loads(response)["status"] != "Complete":
```

Comment:
- It would be good to bound this loop in case due to some unexpected issue/bug we never complete. For example, we can stop after waiting for 3 minutes in total.
- There is no indicator to the user when we are in the loop and doing many iterations to get to a Complete state. Let's add some output. For example, we can say waiting for import status to complete and display a `.` for every iteration.

Similar comment for the export case